staging/src/k8s.io/mount-utils/mount.go

```diff
@@ -147,6 +147,23 @@ func NewMountError(mountErrorValue MountErrorType, format string, args ...interf
 type SafeFormatAndMount struct {
 	Interface
 	Exec utilexec.Interface
+
+	formatSem     chan any
+	formatTimeout time.Duration
+}
+
+// SetMaxConcurrentFormat sets the maximum number of concurrent format
+// operations executed by the mounter. The timeout controls the maximum
+// duration of a format operation before it is treated as complete. A max
+// concurrent value of less than 1 sets concurrency to unlimited.
+// This function assumes that no format operations are currently in flight.
```

Comment:
> This function assumes that no format operations are currently in flight.

should we make this less of an assumption and guard against it?

Reply:
I can't think of a way to ensure safety for in-flight format operations. All [channel axioms](https://dave.cheney.net/2014/03/19/channel-axioms) have rather disastrous implications to the format operations that are attempting to acquire or release semaphores. For example, if I were to close the channel, a goroutine that is actively trying to write to the sem will panic. Similarly, if I create a new channel, a goroutine attempting to read from the sem will hang. Ideally, I would drain the channel, but I can't think of a mechanism.

Reply:
are you expecting callers to set this at initialization time before any mounting is done?

one way to make that obvious is to make the fields public and just let creators set those along with the other fields.

another way to make that obvious is to make this helper function return an error if it is called after mount has been called.

Reply:
Good idea. I created an extensible constructor for drivers that want to use the concurrency limit option.

Reply:
thanks, that's way harder to accidentally misuse, lgtm